commons-statistics-descriptive/src/main/java/org/apache/commons/statistics/descriptive/Variance.java

```diff
@@ -0,0 +1,233 @@
+     * @param value Value.
+     */
+    @Override
+    public abstract void accept(double value);
+
+    /**
+     * Gets the variance of all input values.
+     *
+     * <p>The result is <code>NaN</code> if :
+     * <ul>
+     *     <li>the input array is empty,</li>
+     *     <li>any of the values is <code>NaN</code>, or</li>
+     *     <li>an infinite value of either sign is encountered</li>
+     * </ul>
+     *
+     * <p>The result is <code>0</code> if there is just one finite value in the data set.
+     *
+     * @return {@code Variance} of all values seen so far.
+     */
+    @Override
+    public abstract double getAsDouble();
+
+    /** {@inheritDoc} */
+    @Override
+    public abstract Variance combine(Variance other);
+
+    /**
+     * {@code Variance} implementation that does not store the input value(s) processed so far.
+     */
+    private static class StorelessSampleVariance extends Variance {
+
+        /**
+         * An instance of {@link SumOfSquaredDeviations}, which is used to
+         * compute the variance.
+         */
+        private final SumOfSquaredDeviations squaredDeviationSum;
+
+        /**
+         * Creates a StorelessVariance instance with the sum of squared
+         * deviations from the mean.
+         *
+         * @param squaredDevSum Sum of squared deviations.
+         * @param mean Mean of values.
+         * @param n Number of values.
+         * @param sumOfValues Sum of values.
+         */
+        private StorelessSampleVariance(double squaredDevSum, double mean, long n, double sumOfValues) {
+            squaredDeviationSum = new SumOfSquaredDeviations(squaredDevSum, mean, n, sumOfValues);
+        }
+
+        /**
+         * Create a SumOfSquaredDeviations instance.
+         */
+        StorelessSampleVariance() {
+            squaredDeviationSum = new SumOfSquaredDeviations();
+        }
+
+        /**
+         * Creates a StorelessVariance instance with the sum of squared
+         * deviations from the mean.
+         *
+         * @param squaredDevSum Sum of squared deviations.
+         * @param mean Mean of values.
+         * @param n Number of values.
+         * @param sumOfValues Sum of values.
+         * @return A StorelessVariance instance.
+         */
+        static StorelessSampleVariance create(double squaredDevSum, double mean, long n, double sumOfValues) {
+            return new StorelessSampleVariance(squaredDevSum, mean, n, sumOfValues);
+        }
+
+        @Override
+        public void accept(double value) {
+            squaredDeviationSum.accept(value);
+        }
+
+        @Override
+        public double getAsDouble() {
+            final double sumOfSquaredDev = squaredDeviationSum.getAsDouble();
+            final double n = squaredDeviationSum.getN();
+            if (n == 0) {
+                return Double.NaN;
+            } else if (n == 1 && Double.isFinite(sumOfSquaredDev)) {
+                return 0d;
```

Comment:
No reason to use 0d here. This is only relevant when you wish for the zero to be used as double for double arithmetic, e.g. `1 / 0 != 1 / 0d` (the LHS will throw an arithmetic exception).